frameworks/elastic/src/main/dist/elasticsearch.yml

```diff
@@ -24,4 +24,17 @@ metrics.statsd.port: {{STATSD_UDP_PORT}}
 gateway.expected_master_nodes: 3
 gateway.expected_data_nodes: {{DATA_NODE_COUNT}}
 gateway.recover_after_master_nodes: 3
-gateway.recover_after_data_nodes: {{DATA_NODE_COUNT}}
\ No newline at end of file
+gateway.recover_after_data_nodes: {{DATA_NODE_COUNT}}
+{{#ELASTIC_ENABLE_TLS}}
```

Comment:
Are you sure this section does not appear when you have TLS disabled? I'd expect it to always show up because of https://jira.mesosphere.com/browse/INFINITY-2268.

Reply:
@mhrabovcin @nickbp What's the deal with this?

Reply:
I've tried to start Elastic from this PR without TLS the "if" block was behaving as expected and security configuration wasn't rendered.

```
2017/09/08 14:35:23 Writing rendered 'elasticsearch-5.5.2/config/elasticsearch.yml' from envvar 'CONFIG_TEMPLATE_ELASTICSEARCH' with the following changes (1533 bytes -> 837 bytes):
  discovery.zen.ping.unicast.hosts:
-   - "master-0-node.{{FRAMEWORK_HOST}}:{{MASTER_NODE_TRANSPORT_PORT}}"
-   - "master-1-node.{{FRAMEWORK_HOST}}:{{MASTER_NODE_TRANSPORT_PORT}}"
-   - "master-2-node.{{FRAMEWORK_HOST}}:{{MASTER_NODE_TRANSPORT_PORT}}"
+   - "master-0-node.elastic.autoip.dcos.thisdcos.directory:9300"
+   - "master-1-node.elastic.autoip.dcos.thisdcos.directory:9300"
+   - "master-2-node.elastic.autoip.dcos.thisdcos.directory:9300"
  discovery.zen.minimum_master_nodes: 2
  network.bind_host:
-   - {{MESOS_CONTAINER_IP}}
+   - 10.0.3.65
    - _local_
  network.publish_host:
-   - {{TASK_NAME}}.{{FRAMEWORK_HOST}}
+   - master-1-node.elastic.autoip.dcos.thisdcos.directory
    - _local_
- cluster.name: {{CLUSTER_NAME}}
- node.name: {{TASK_NAME}}
+ cluster.name: elastic
+ node.name: master-1-node
  path.data: "container-path/data"
  path.logs: "container-path/logs"
- http.port: {{PORT_HTTP}}
- transport.tcp.port: {{PORT_TRANSPORT}}
+ http.port: 1027
+ transport.tcp.port: 9300
  bootstrap.memory_lock: false
- node.master: {{MASTER_ENABLED}}
- node.data: {{DATA_ENABLED}}
- node.ingest: {{INGEST_ENABLED}}
- metrics.statsd.host: {{STATSD_UDP_HOST}}
- metrics.statsd.port: {{STATSD_UDP_PORT}}
+ node.master: true
+ node.data: false
+ node.ingest: false
+ metrics.statsd.host: 198.51.100.1
+ metrics.statsd.port: 33806
  gateway.expected_master_nodes: 3
- gateway.expected_data_nodes: {{DATA_NODE_COUNT}}
+ gateway.expected_data_nodes: 2
  gateway.recover_after_master_nodes: 3
- gateway.recover_after_data_nodes: {{DATA_NODE_COUNT}}
- {{#ELASTIC_ENABLE_TLS}}
- # https://www.elastic.co/guide/en/elasticsearch/reference/5.5/security-settings.html#ssl-tls-settings
- xpack.security.enabled: true
- xpack.ssl.key: {{MESOS_SANDBOX}}/node.key
- xpack.ssl.certificate: {{MESOS_SANDBOX}}/node.crt
- xpack.ssl.certificate_authorities: [ "{{MESOS_SANDBOX}}/node.ca" ]
- xpack.ssl.client_authentication: optional
- # Node to node encryption
- xpack.security.transport.ssl.enabled: true
- # Client encryption
- xpack.security.http.ssl.enabled: true
- xpack.security.http.ssl.client_authentication: none
- {{/ELASTIC_ENABLE_TLS}}
+ gateway.recover_after_data_nodes: 2
+ 
```

Reply:
¯\_(ツ)_/¯  I don't get why it works, but it sure looks like it works!